module4/homework/appendNewRecipe/AppendNewRecipe.cpp

```diff
@@ -0,0 +1,51 @@
+#include "AppendNewRecipe.hpp"
+
+#include <algorithm>
+#include <map>
+const std::map<char, std::string> ingredientsUnits{{'g', "gram"}, {'s', "szklanka(i)"}, {'m', "mililitrow"}};
+
+std::vector<std::string> FormatIngredients(const std::list<std::string>& ingredients,
+                                           const std::deque<std::pair<size_t, char>>& amount) {
+    std::vector<std::string> ingredientsResult;
+    std::transform(ingredients.begin(), ingredients.end(), amount.begin(), std::back_inserter(ingredientsResult),
+                   [&](const auto& ingredient, const auto& amount) {
+                       std::stringstream ingredientsStream;
+                       ingredientsStream << amount.first << ' ' << ingredientsUnits.at(amount.second) << ' '
+                                         << ingredient;
+                       return ingredientsStream.str();
+                   });
+    return ingredientsResult;
+}
+
+std::stringstream FormatRecipit(std::vector<std::string> steps,
+                                const std::list<std::string>& ingredients,
+                                const std::deque<std::pair<size_t, char>>& amount) {
+    std::stringstream recipeStream;
+
+    recipeStream << "Skladniki:\n";
```

Comment:
As someone mentioned on one of previous PRs, you can move it to variable

Reply:
Ok, thanks.

Reply:
+1